src/middleware/auth.middleware.ts

```diff
@@ -0,0 +1,10 @@
+import { Action } from 'routing-controllers';
+import { decodeJWT } from '@services/jwt.service';
+
+const authorizationChecker = async (action: Action): Promise<boolean> => {
+  const token = action.request.headers['authorization'];
+  const payload = decodeJWT(token);
+  return payload !== null;
```

Comment:
Maybe we should fetch a value from the payload instead of just comparing it with null 🤔. I think the decode just return the object if it's valid... note: Test with an expired token

Reply:
I have a few questions there @Alxzu: what happens if the token is not set in header or null? If the decoded token is expired, the payload will be null anyways? At some point, we will want to store expired tokens in the database, after that these functions should make some checks to contemplate 'logouts', maybe something like this:

```
// Check if the token was stored as expired
const isExpired = await this.expiredTokenRepository.findOne({
		where: {
			token: token
		}
});
if (isExpired) {
	throw new HttpErrors.Unauthorized("This token is no longer active");
}
```

Reply:
Totally agree, we want to have a mechanism to revoke tokens. I think we can handle that on another PR. This is already too big and I would prefer having a basic setup before moving into more logic.